src/Symfony/Contracts/HttpClient/HttpClientInterface.php

```diff
@@ -19,6 +19,8 @@
  *
  * @see HttpClientTestCase for a reference test suite
  *
+ * @method static withOptions(array $options) Returns a new instance of the client with new default options
```

Comment:
This is not pretty. :/

Reply:
`DebugClassLoader` will use this to trigger deprecations when implements don't have the method.

Reply:
Why withOptions() is described as static? Is it intended? Everywhere it is implemented as normal method.

Reply:
`static` is the return type. See #42214 or #42358 for previous discussions on the topic.